### PR TITLE
[Client encryption]: Fixes Faster validations and string comparisons

### PR DESCRIPTION
**Pull request by JanHyka**
# Pull Request Template

## Description

1. Switch frequent constants string comparison mode to `ordinal`
2. Reduce allocations and validations complexity

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)

## Closing issues

Contributes to #4678

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionProcessor.cs [hunk 1]
```diff
@@ -55,26 +55,26 @@ public static async Task<Stream> EncryptAsync(
                 return input;
             }
 
-            if (!encryptionOptions.PathsToEncrypt.Distinct().SequenceEqual(encryptionOptions.PathsToEncrypt))
+            if (encryptionOptions.PathsToEncrypt.Distinct().Count() != encryptionOptions.PathsToEncrypt.Count())
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionProcessor.cs:58]: SequenceEquals covering values right? where as new change is just the count not values?

Comment (JanHyka) [reply Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionProcessor.cs:58]: We are comparing encryptionOptions.PathsToEncrypt on both sides... that means values are always identical and is useless to compare them. We only need to validate there are no repetitions - ie. Distinct().Count() vs Count()

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionProcessor.cs:58]: You are right its the same argument. I mis read it before. Thanks for clarifying it.

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**